hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/DeleteTableProcedure.java

```diff
@@ -348,6 +347,13 @@ protected static void deleteFromFs(final MasterProcedureEnv env,
         throw new IOException("Couldn't delete mob dir " + mobTableDir);
       }
     }
+
+    // Delete the directory on wal filesystem
+    FileSystem walFs = mfs.getWALFileSystem();
+    Path tableWALDir = FSUtils.getWALTableDir(env.getMasterConfiguration(), tableName);
+    if (walFs.exists(tableWALDir) && !walFs.delete(tableWALDir, true)) {
+      throw new IOException("Couldn't delete table dir on wal filesystem" + tableWALDir);
+    }
```

Comment:
Should we not be checking for wrong wal directory here too and deleting that if it exists?

Reply:
Not necessary I think, at least there is no way to delete the namespace directories... So the users hava to use a script to delete the empty directories manually...